data/rulesets.ts

```diff
@@ -735,6 +735,27 @@ export const BattleFormats: {[k: string]: FormatsData} = {
 			}
 		},
 	},
+	stadiumsleepclausemod: {
+		effectType: 'Rule',
+		name: 'Stadium Sleep Clause Mod',
+		desc: "Prevents players from putting one of their opponent's Pok&eacute;mon to sleep if any of the opponent's other Pok&eacute;mon are asleep",
+		onBegin() {
+			this.add('rule', 'Stadium Sleep Clause Mod: Limit one foe put to sleep');
+		},
+		onSetStatus(status, target, source) {
+			if (source && source.side === target.side) {
+				return;
+			}
```

Comment:
Wait, I thought you said it activates on Rest? Why do you still have the part of the code that stops it from activating from Rest?

Reply:
Because Rest also calls onSetStatus, and if I don't have that check to see if it's coming from your own side, Sleep Clause  would prevent the user from using Rest if any other Pokemon on the same side is asleep, which as far as I am aware should not occur.

Reply:
Wait, that's the opposite of what you said in the intro:

> This means if a player's Pokemon uses Rest, Sleep Clause is activated for their side and is protected from Sleep moves, even though the sleep was self-inflicted from Rest.

Reply:
Oh, I see, the difference is that you can still use Rest on yourself after you've already Rested, but your opponent can't use sleep moves on you.

I think that should probably be documented clearly in the mod description.